hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -177,6 +179,16 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from nodepool: %w", err)
 			}
 		}
+
+		hcluster, err := GetHostedClusterByName(ctx, r.Client, nodePool.GetNamespace(), nodePool.Spec.ClusterName)
+		if !apierrors.IsNotFound(err) { // maybe already deleted
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("can't get HostedCluster %s/%s: %w", nodePool.GetNamespace(), nodePool.Spec.ClusterName, err)
+			}
+			// found
+			r.KubevirtInfraClients.Delete(hcluster.Spec.InfraID)
+		}
+
```

Comment:
I think we can simplify this so there's no dependency on the HC during cleanup of the NP.

Use something unique on the nodepool as the key for the client. the nodepool.ObjectMeta.UID seems like it would work fine.